Google.PowerShell/PubSub/GcpsTopic.cs

```diff
@@ -0,0 +1,234 @@
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [Alias("Name")]
+        public string[] Topic { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (Topic != null && Topic.Length > 0)
+            {
+                foreach (string topicName in Topic)
+                {
+                    string formattedTopicName = PrefixProjectToTopic(topicName, Project);
+                    try
+                    {
+                        ProjectsResource.TopicsResource.GetRequest getRequest = Service.Projects.Topics.Get(formattedTopicName);
+                        WriteObject(getRequest.Execute());
+                    }
+                    catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+                    {
+                        ErrorRecord errorRecord = new ErrorRecord(
+                            new ItemNotFoundException($"Topic '{topicName}' does not exist in project '{Project}'."),
+                            "TopicNotFound",
+                            ErrorCategory.ObjectNotFound,
+                            topicName);
+                        WriteError(errorRecord);
+                    }
+                }
+            }
+            else
+            {
+                ProjectsResource.TopicsResource.ListRequest listRequest = Service.Projects.Topics.List($"projects/{Project}");
+                do
+                {
+                    ListTopicsResponse response = listRequest.Execute();
+
+                    if (response.Topics != null)
+                    {
+                        WriteObject(response.Topics, true);
+                    }
+                    listRequest.PageToken = response.NextPageToken;
+                }
+                while (!Stopping && listRequest.PageToken != null);
+            }
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Removes Google Cloud PubSub topics.
+    /// </para>
+    /// <para type="description">
+    /// Removes one or more Gooogle Cloud PubSub topics. Will throw errors if the topics do not exist.
+    /// The cmdlet will delete the topics in the default project if -Project is not used.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Remove-GcpsTopic -Topic "my-topic"</code>
+    ///   <para>This command removes topic "my-topic" in the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Remove-GcpsTopic -Topic "topic1", "topic2" -Project "my-project"</code>
+    ///   <para>
+    ///   This command removes 2 topics ("topic1" and "topic2") in the project "my-project".
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/pubsub/docs/publisher#delete)">
+    /// [Deleting a Topic]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "GcpsTopic")]
```

Comment:
Since this deletes data, we should add `SupportsShouldProcess = true` and then call `ShouldProcess` depending on the Force/Confirm flag.